lib/cubit/nwc/nwc_cubit.dart

```diff
@@ -0,0 +1,151 @@
+import 'dart:async';
+import 'package:breez_sdk_liquid/breez_sdk_liquid.dart';
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:logging/logging.dart';
+
+import 'package:misty_breez/cubit/nwc/nwc_state.dart';
+
+final Logger _logger = Logger('NwcCubit');
+
+class NwcCubit extends Cubit<NwcState> {
+  final BreezSDKLiquid breezSdkLiquid;
+
+  NwcCubit(this.breezSdkLiquid) : super(NwcState.initial()) {
+    loadConnections();
+  }
+
+  /// Loads all NWC connections from the service
+  Future<void> loadConnections() async {
+    emit(state.copyWith(isLoading: true));
+
+    try {
+      final BreezNwcService? nwcService = breezSdkLiquid.plugins?.nwc;
+
+      if (nwcService == null) {
+        emit(state.copyWith(connections: <NwcConnectionModel>[], isLoading: false, error: 'NWC service is not available'));
+        return;
+      }
+
+      final Map<String, NwcConnection> connectionsMap = await nwcService.listConnections();
+
+      final List<NwcConnectionModel> connections =
+          connectionsMap.entries
+              .map(
+                (MapEntry<String, NwcConnection> entry) => NwcConnectionModel(
+                  name: entry.key,
+                  connectionString: entry.value.connectionString,
+                  periodicBudget: entry.value.periodicBudget,
+                  expiryTimeSec: entry.value.expiryTimeSec,
+                  createdAt: entry.value.createdAt,
+                ),
+              )
+              .toList()
+            ..sort(
+              (NwcConnectionModel a, NwcConnectionModel b) =>
+                  a.name.toLowerCase().compareTo(b.name.toLowerCase()),
+            );
+
+      emit(state.copyWith(connections: connections, isLoading: false));
```

Comment:
```suggestion
      emit(NwcState(connections: connections));
```

This creates a fresh `NwcState` instead of copying the previous one, which should help prevent unintended state persistence.